packages/fluentui/react-northstar/src/types.ts

```diff
@@ -64,7 +64,13 @@ type ReactNode = React.ReactChild | React.ReactNodeArray | React.ReactPortal | b
 
 export type ShorthandValue<P extends Props> = ReactNode | ObjectShorthandValue<P>;
 
-export type ShorthandCollection<P, K = never> = ShorthandValue<P & { kind?: K }>[];
+type KindSelector<T> = {
+  [P in keyof T]: { kind: P } & T[P];
+}[keyof T];
+
+export type ShorthandCollection<Props, Kinds = Record<string, {}>> = ShorthandValue<
+  Props | (KindSelector<Kinds> & Props)
```

Comment:
```suggestion
  Props | KindSelector<Kinds>
```

Is there any need in `& Props`?

Reply:
Yep, without it would break in the `Toolbar` component in:

```typescript
{ToolbarItem.create(overflowItem, {
          defaultProps: () => ({
            icon: <MoreIcon outline />,
          }),
          overrideProps: {
            menu: { items: overflowOpen ? collectOverflowItems() : [], popper: { positionFixed: true } },
            menuOpen: overflowOpen,
            onMenuOpenChange: (e, { menuOpen }) => {
              _.invoke(props, 'onOverflowOpenChange', e, { ...props, overflowOpen: menuOpen });
            },
          },
        })}
```